File: platforms/ios/runtime/Sources/NativeblocksRuntime/logger/LoggerAdapter.swift
import Foundation
import NativeblocksRuntimeFFI

internal final class LoggerAdapter: NativeblocksRuntimeFFI.Logger, @unchecked Sendable {

    private let delegate: any INativeLogger

    init(delegate: any INativeLogger) {
        self.delegate = delegate
    }

    func log(
        level: RuntimeFFILoggerEventLevel,
        event: String,
        message: String,
        parameters: [String: String]
    ) {
        delegate.logWithContext(
            level: level.toDomain(),
            event: event,
            message: message,
            parameters: parameters
        )
    }
}

File: platforms/android/runtime/core/src/main/java/io/nativeblocks/runtime/logger/LoggerAdapter.kt
package io.nativeblocks.runtime.logger

import io.nativeblocks.runtime.api.provider.logger.INativeLogger
import io.nativeblocks.runtime.engine.Logger
import io.nativeblocks.runtime.engine.LoggerEventLevel
import io.nativeblocks.runtime.engine.toDomain

internal class LoggerAdapter(
    private val delegate: INativeLogger,
) : Logger {

    override fun log(
        level: LoggerEventLevel,
        event: String,
        message: String,
        parameters: Map<String, String>,
    ) {
        delegate.log(
            level = level.toDomain(),
            event = event,
            message = message,
            parameters = parameters,
        )
    }
}